src/plot_api/helpers.js

```diff
@@ -299,7 +299,7 @@ exports.cleanData = function(data, existingData) {
             trace.scene = Plots.subplotsRegistry.gl3d.cleanId(trace.scene);
         }
 
-        if(!Registry.traceIs(trace, 'pie')) {
+        if(!Registry.traceIs(trace, 'pie') && !Registry.traceIs(trace, 'bar')) {
```

Comment:
That's what this line change is meant for; i.e. to exclude bar's `textposition` from those mappings.